src/main/java/me/zeroeightsix/kami/module/modules/movement/ElytraFlight.kt

```diff
@@ -182,38 +202,134 @@ class ElytraFlight : Module() {
         } else elytraDurability = 0
 
         /* Holds player in the air if run out of durability */
-        if (!mc.player.onGround && outOfDurability && elytraDurability <= 1) {
-            event.cancel()
-            mc.player.setVelocity(0.0, -0.01, 0.0)
+        if (!mc.player.onGround && elytraDurability <= 1 && outOfDurability) {
+            holdPlayer(event)
         } else if (outOfDurability) outOfDurability = false /* Reset if players is on ground or replace with a new elytra */
 
         /* Elytra flying status check */
         isFlying = mc.player.isElytraFlying || (mc.player.capabilities.isFlying && mode.value == ElytraFlightMode.CREATIVE)
         isPacketFlying = isPacketFlying && mode.value == ElytraFlightMode.PACKET
 
         /* Movement input check */
-        if (isFlying) {
-            isStandingStillH = mc.player.movementInput.moveForward == 0f && mc.player.movementInput.moveStrafe == 0f
-            isStandingStill = isStandingStillH && !mc.player.movementInput.jump && !mc.player.movementInput.sneak
-        } else isStandingStill = false
+        isStandingStillH = mc.player.movementInput.moveForward == 0f && mc.player.movementInput.moveStrafe == 0f
+        isStandingStill = isStandingStillH && !mc.player.movementInput.jump && !mc.player.movementInput.sneak
+
+        /* Reset acceleration */
+        if (!isFlying || isStandingStill) speedPercentage = accelerateStartSpeed.value.toFloat()
+
+        /* Remove legs swing */
+        if (autoLanding.value || isFlying && (mode.value == ElytraFlightMode.CONTROL || mode.value == ElytraFlightMode.PACKET)) {
+            mc.player.prevLimbSwingAmount = 0.0f
+            mc.player.limbSwing = 0.0f
+            mc.player.limbSwingAmount = 0.0f
+        }
+    }
+
+    /* Holds player in the air */
+    private fun holdPlayer(event: PlayerTravelEvent) {
+        event.cancel()
+        mc.timer.tickLength = 50.0f
+        mc.player.setVelocity(0.0, -0.01, 0.0)
+    }
+
+    /* Auto landing */
+    private fun landing(event: PlayerTravelEvent) {
+        when {
+            mc.player.onGround -> {
+                sendChatMessage("$chatName Landed!")
+                autoLanding.value = false
+                return
+            }
+            MODULE_MANAGER.getModuleT(LagNotifier::class.java).takeoffPaused -> {
+                holdPlayer(event)
+            }
+            checkForLiquid() -> {
```

Comment:
shouldn't this check come before holding the player? if they can't land anyways, why hold?